src/Orleans.Core.Abstractions/Async/PublicOrleansTaskExtensions.cs

```diff
@@ -31,7 +31,7 @@ public static void Ignore(this Task task)
                 task.ContinueWith(
                     IgnoreTaskContinuation,
                     CancellationToken.None,
-                    TaskContinuationOptions.NotOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
+                    TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously,
```

Comment:
Why this change?

Reply:
To avoid running the continuation if the task is canceled, just more optimal.

Reply:
I ran some tests and the behavior seems fine